src/components/KCopy/KCopy.vue

```diff
@@ -0,0 +1,263 @@
+<template>
+  <div class="k-copy">
+    <span
+      v-if="badge"
+      class="copy-badge-text"
+    >
+      {{ badgeText }}
+    </span>
+    <div
+      :class="[
+        'copy-container',
+        truncateStyles,
+        badgeStyles
+      ]"
+    >
+      <KTooltip
+        v-if="format !== 'hidden'"
+        class="copy-text"
+        :class="[truncateContent, useMono]"
+        data-testid="copy-text"
+        :label="textTooltip"
+        placement="bottomStart"
+        position-fixed
+      >
+        <span>
```

Comment:
question: what's the purpose of the `span` tag here?

Reply:
Added it just to provide a way to be able to access that text in some unforeseeable case it needs some styling change through host app. 
<img width="669" alt="image 1644" src="https://github.com/Kong/kongponents/assets/87779967/76f2345a-252c-446d-9c81-28b5b7ea4d04">

Reply:
I don't think it's needed; the host app shouldn't be overriding styles

Reply:
span removed.

Reply:
Need a div here, otherwise the KTooltip won't display on hover.